### PR TITLE
use reexports to avoid specifying submodules in many cases

### PR DESCRIPTION
**Pull request by PenguinWithATie**
use reexports to avoid specifying submodules in many cases

also some tailwind classes got moved due to my tailwind sorter plugin, I figure it doesn't hurt to have them also but let me know if I should remove them

### CODE REVIEW

File: engine/src/lib.rs [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
+mod player;
+mod position;
+mod state;
+mod torus_array;
+mod turn;
+pub use board::Board;
+pub use bug::Bug;
+pub use bug_stack::BugStack;
+pub use color::{Color, ColorChoice};
+pub use dfs_info::DfsInfo;
```
Comment (klautcomputing) [inline engine/src/lib.rs:25]: `DfsInfo` and `MidMoveBoard` are internal types and it should not be accessible from the outside.
Unsure about the `TorusArray`...

**Review by klautcomputing — CHANGES_REQUESTED** [review]
Thanks for your PR! Looks a lot cleaner now, I just left a comment about 2-3 types that are internal and shouldn't be available from the outside.

File: engine/src/lib.rs (hunk 1 above)
Comment (PenguinWithATie) [reply engine/src/lib.rs]: You are correct, none of these types are required to be public, the Hasher type as well

**Review by klautcomputing — APPROVED** [review]
:shipit:

**Comment by klautcomputing** [comment]
Can you make a single commit to be rebased into main, please?

**Review by klautcomputing — APPROVED** [review]
Thank you for this!

**Review by klautcomputing — APPROVED** [review]
🚢 🇮🇹

**Pull request merged**